ui/pages/confirm-transaction-base/gas-details-item/gas-details-item.js

```diff
@@ -0,0 +1,136 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { COLORS } from '../../../helpers/constants/design-system';
+import { PRIMARY, SECONDARY } from '../../../helpers/constants/common';
+import { hexWEIToDecGWEI } from '../../../helpers/utils/conversions.util';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+
+import Box from '../../../components/ui/box';
+import Typography from '../../../components/ui/typography/typography';
+import GasTiming from '../../../components/app/gas-timing/gas-timing.component';
+import I18nValue from '../../../components/ui/i18n-value';
+import InfoTooltip from '../../../components/ui/info-tooltip/info-tooltip';
+import LoadingHeartBeat from '../../../components/ui/loading-heartbeat';
+import TransactionDetailItem from '../../../components/app/transaction-detail-item/transaction-detail-item.component';
+import UserPreferencedCurrencyDisplay from '../../../components/app/user-preferenced-currency-display';
+
+const HeartBeat = () =>
+  process.env.IN_TEST === 'true' ? null : <LoadingHeartBeat />;
+
+const GasDetailItem = ({
+  hexMaximumTransactionFee,
+  hexMinimumTransactionFee,
+  isMainnet,
+  maxFeePerGas,
+  maxPriorityFeePerGas,
+  supportsEIP1559,
+  txData,
+  useNativeCurrencyAsPrimaryCurrency,
+}) => {
+  const t = useI18nContext();
+  return (
+    <TransactionDetailItem
+      key="gas-item"
+      detailTitle={
+        <Box display="flex">
+          <Box marginRight={1}>
+            <I18nValue messageKey="transactionDetailGasHeadingV2" />
+          </Box>
+          <span className="gas-details-item__estimate">
+            (<I18nValue messageKey="transactionDetailGasInfoV2" />)
+          </span>
+          <InfoTooltip
+            contentText={
+              <>
+                <Typography fontSize="12px">
+                  {t('transactionDetailGasTooltipIntro', [
```

Comment:
[Q]: A little curious on why we can't use `<I18nValue messageKey ="" />` here and in a couple of following locations in this file if we are to introduce the new component `<I18nValue />`?

Reply:
Good point, I actually tried that but at some places it does not work. To me reason looks appears to be issue in accessing transaction object from  context.